### DISCUSSION TITLE
Web-UI Lite

### DISCUSSION THREAD

**Discussion by RUHavingFun** [mrlt8/docker-wyze-bridge#567] [Ideas]
My favorite new feature is the Web-UI.  But I have one request that would truly make it my most used feature.  A lite mode.  Or in other words a mode where it was nothing but the camera images with very little to no border.

I currently run a Raspberry Pi 3B+ with Android of Things running on it so that I can run TinyCam and show my 4 main cameras on screen at once.  I would love to simplify and just run Raspbian and the Web-UI.  But the camera images are too far apart to work as desired.

![IMG_6974](https://user-images.githubusercontent.com/39495965/197203277-2c23d634-4fd9-492c-8c4e-4cba952238f8.jpeg)

Anyway - just a thought.  Keep up the good work.  This has been an amazing project that has changed how I use my cameras.

**Comment by mrlt8** [comment]
Maybe like a "full screen" mode that hides all the buttons and extras?

**Reply by RUHavingFun** [reply]
Kinda - but that can already be achieved with single cam feeds.  I am looking for where you can display multiple cameras at once.

**Reply by mrlt8** [reply]
I was thinking a full screen toggle on the multi-cam page that hides all the extras and changes the css to fill the display while still showing all the cameras.

**Reply by RUHavingFun** [reply]
Perfect!

**Reply by capncybo** [reply]
Changing the Order of the Camera Feeds would also be REALLY nice.  I know that by using filtering I can Hide a camera or two... But I have not "yet" found a way to re-arrange or change the order of the camera feeds.  Which really annoys the hell out of me + It would also be great if we could set a column default such as THREE for example.  The Auto-sizing default is NOT what I want.

**Reply by cheme75** [reply]
> Changing the Order of the Camera Feeds would also be REALLY nice. I know that by using filtering I can Hide a camera or two... But I have not "yet" found a way to re-arrange or change the order of the camera feeds. Which really annoys the hell out of me + It would also be great if we could set a column default such as THREE for example. The Auto-sizing default is NOT what I want.

I’ve been able to click and drag the cams to position them as desired in the web ui and that positioning seems to remain from session to session, even if it didn’t, it only takes seconds to click and drag - thrills the hell out of me! MAYBE try to click, hold and drag to arrange. Not sure what the default setting is but my 2 column layout IS retained. If you are asking for a feature to resize each feed, then I’ve not yet seen that. I find @mrlt8 has been extraordinarily responsive when his time allows.

**Reply by capncybo** [reply]
Thanks but I should have specified... I already knew about the clicking & dragging to re-order windows after start-up.  What I meant to request was actually the ability to pre-configure the camera-streams order & number of columns.  You're fortunate if you prefer 2 columns as that is the default.  On my monitor I'd prefer 3 but it keeps changing back into two.  The resizing seems somewhat automated & tied to the number of columns.  Anyways it was just a request.  IMO the re-ordering manually after bringing up the web-page "each-time" kinda sucks.  The user should be able to determine the order before-hand.  Didn't mean to complain so much, sorry.  I'm just hoping it can be implemented.

**Reply by RUHavingFun** [reply]
@cheme75 Where are you clicking and dragging?  What browser are you using?  --  Never mind - I found it.  That was not obvious.  It is only on the title bar area of each camera to the right of the name and before the details settings button.  Your cursor turns into a move cursor.  Nice tip.  It does seem to be locally cached but not remember across browsers or after full close and refresh.

**Reply by capncybo** [reply]
I typically use Firefox & can re-order the camera windows By moving the mouse pointer to somewhere around the middle of the windows' title-bar & the right-most corner... Click & holding then dragging the stream elsewhere & releasing... However if I exit the browser completely & relaunch all changes are lost it reverts back to two columns with the original order (of which I don't like).
**EDIT** sorry you are a quicker typist than me by far LOL

**Reply by cheme75** [reply]
> I typically use Firefox & can re-order the camera windows By moving the mouse pointer to somewhere around the middle of the windows' title-bar & the right-most corner... Click & holding then dragging the stream elsewhere & releasing... However if I exit the browser completely & relaunch all changes are lost it reverts back to two columns with the original order (of which I don't like).
> **EDIT** sorry you are a quicker typist than me by far LOL

Generally use Firefox and it has kept my order of location but I’ve not tried closing browser and re-opening to test retention, only leaving page, returning or minimizing browser and reopening. Now you mention, using it in safari on iPad I do recall reordering the windows. Maybe he can set a cookie to retain position info. 

Wasn’t sure on columns, only got 4 cams/d bells so the 2 columns looks good. I tried 3-4 but 2 looked better. Wasn’t sure if it was default. I bet he can a retention options that reads the cookie settings or some other newer method.

**Comment by mrlt8** [comment]
Pushed some changes to the [dev branch](https://github.com/mrlt8/docker-wyze-bridge/pull/569) that should allow switching to a more basic kiosk/fullscreen view. Let me know what you guys think!

@capncybo changes to the web-ui should get saved to a cookie so I'm not sure why they keep getting reset for you. I did add an option to make changes via query parameters, so you should be able to bookmark your setting e.g. `?columns=3&refresh=100&order=garage,porch&fullscreen`

**Reply by RUHavingFun** [reply]
Works perfectly!  Thanks.

**Reply by capncybo** [reply]
@mrlt8 Thank you ever so-much, sorry I was outta-town & missed the chance to test. However I just noticed it's already made it into the latest version so... thanks again & well-done!

**Reply by capncybo** [reply]
@mrlt8 IMO it's very **close** to perfect, but what would REALLY be awesome is...
Supporting MULTIPLE "series" Query params such as the following examples...
http://localhost:5000/?fullscreen?columns=4
http://localhost:5000/?columns=4?order=front-cam,back-cam,garage,other
Note: I have already tried playing with these & it seems ONLY the first parameter is implemented

**Reply by mrlt8** [reply]
hmm... multiple queries _seem_ to be working for me on macOS (safari, chrome, firefox).
```
http://localhost:5000/?video&columns=5&fullscreen&order=back
```
Can you post your browser and the exact url you're trying to use?

**Reply by capncybo** [reply]
@mrlt8 Thanks for the additional example. (Please notice in my two examples above, I was using multiple "?")
In your example there is ONE ? after the server but there are actually "&" in-between all the parameters.  So I take it back... It works perfectly in firefox I tested using: http://localhost:5000/?fullscreen&columns=5&order=front-door,front-yard,alley,garage
Suggestion: Maybe add one of these more advanced URL as an example
Thanks again, Fantastic Job!!!
